src/sagemaker/clarify.py

```diff
@@ -2236,23 +2481,45 @@ def _add_methods(
     def _merge_explainability_configs(
         cls,
         explainability_config: Union[ExplainabilityConfig, List[ExplainabilityConfig]],
+        time_series_case: bool = False,
     ):
         """Merges explainability configs, when more than one."""
         if isinstance(explainability_config, list):
             explainability_methods = {}
             if len(explainability_config) == 0:
                 raise ValueError("Please provide at least one explainability config.")
             for config in explainability_config:
+                # ensure all provided explainability configs
+                # are AsymmetricSHAPConfig in time series case
+                is_asym_shap_config = isinstance(config, AsymmetricSHAPConfig)
+                if time_series_case and not is_asym_shap_config:
+                    raise ValueError(
+                        "Please provide only Asymmetric SHAP configs for TimeSeries explainability."
+                    )
+                if not time_series_case and is_asym_shap_config:
+                    raise ValueError(
+                        "Please do not provide Asymmetric SHAP configs for non-TimeSeries uses."
+                    )
                 explain_config = config.get_explainability_config()
                 explainability_methods.update(explain_config)
             if not len(explainability_methods) == len(explainability_config):
                 raise ValueError("Duplicate explainability configs are provided")
             if (
-                "shap" not in explainability_methods
+                not time_series_case
+                and "shap" not in explainability_methods
                 and "features" not in explainability_methods["pdp"]
             ):
                 raise ValueError("PDP features must be provided when ShapConfig is not provided")
             return explainability_methods
+        is_asym_shap_config = isinstance(explainability_config, AsymmetricSHAPConfig)
```

Comment:
Thought:

I see you keep checking the same (or similar) things again and again. This makes the code problematic to read, test, and maintain. 

Let's simplify this by having the input validation earlier and once for all (without extreme though)

Reply:
This is a separate check that would not occur if the function ever entered its loop but as we've discussed `_merge_explainability_config` needs to be modified anyway.